packages/target-ethers-v5/src/index.ts

```diff
@@ -129,6 +133,20 @@ export default class Ethers extends TsGeneratorPlugin {
     return allFiles
   }
 
+  private genCommons(): string {
+    return `
+    import { EventFilter, Event } from 'ethers'
+    import { Result } from '@ethersproject/abi'
+
+    export interface TypedEventFilter<T, G> extends EventFilter {}
+
+    export interface TypedEvent<T extends Result> extends Event {
+      args: T;
+    }
+    
+    export type TypedListener<T extends Array<any>, G> = (...listenerArg: [...T, TypedEventFilter<T, G>]) => void;`
```

Comment:
Can we use whole word type names instead of just `T` & `G`? I didn't use ethers events for some time so now I don't know what are these args without refering to ethers docs (and frankly the docs don't help much either 😅)

Reply:
The `event.args` in ethers js allows access as a array or by parameter name. For e.g. `Transfer(address,address,uint256)` the queryFilter will output the following interfce:

```ts
import { Event } from 'ethers'

interface TransferEvent extends Event {
  args: [string, string, BigNumber] & { from: string, to: string, value: BigNumber }
}
```

So there are two components `[string, string, BigNumber]` and `{ from: string, to: string, value: BigNumber }`. Initially I was not separating them out, but the `contract.on` needs the first component to be able to have types on the callback.

I agree they should be renamed instead of leaving T & G. What do you think about `T` =>`ArrayComponent` and `G` => `ObjectComponent` or any better ideas?